include/mata/inter-aut.hh

```diff
@@ -0,0 +1,193 @@
+ * and type of alphabet. It contains also the transitions formula and formula for initial and final
+ * states. The formulas are represented as tree where nodes are either operands or operators.
+ */
+struct InterAutomaton
+{
+    /**
+     * Type of automaton. So far we support nondeterministic finite automata (NFA) and
+     * alternating finite automata (AFA)
+     */
+    enum AutomatonType
+    {
+        NFA,
+        AFA
+    };
+
+    /**
+     * The possible kinds of naming of items in sets of states, nodes, or symbols.
+     * It implies how the given set is defined.
+     * Naming could be automatic (all things in formula not belonging to other sets will be assigned to a
+     * set with automatic naming), marker based (everything beginning with `q` is a state, with `s` is a symbol,
+     * with `n` is a node), or enumerated (the given set is defined by enumeration).
+     */
+    enum Naming
+    {
+        AUTO,
+        MARKER,
+        ENUM
+    };
+
+    /**
+     * Type of alphabet. We can have explicit alphabet (containing explicit symbols), or bitvector, or class of symbols
+     * (e.g., alphabet is everything in utf), or intervals.
+     * So far, only explicit representation is supported.
+     */
+    enum AlphabetType
+    {
+        EXPLICIT,
+        BITVECTOR,
+        CLASS,
+        INTERVALS
+    };
+
+public:
+    Naming state_naming;
+    Naming symbol_naming;
+    Naming node_naming;
+    AlphabetType alphabet_type;
+    AutomatonType automaton_type;
+
+    // The vectors represent the given sets when enumeration is used.
+    std::vector<std::string> states_names;
+    std::vector<std::string> symbols_names;
+    std::vector<std::string> nodes_names;
+
+    FormulaGraph initial_formula;
+    FormulaGraph final_formula;
+
+    /**
+     * Transitions are paris where the first member is left handed side of transition (i.e., a state)
+     * and the second item is a graph representing transition formula (which can contain symbols, nodes, and states).
+     */
+    struct std::vector<std::pair<FormulaNode, FormulaGraph>> transitions;
+
+    /**
+     * A method for building a vector of InterAutomaton for a parsed input.
+     * For each section in input is created one InterAutomaton.
+     * It parses basic information about type of automata, its naming conventions etc.
+     * Then it parses input and final formula of automaton.
+     * Finally, transition formulas are transformed to graph representation by
+     * turning an input stream of tokens to postfix notation and then
+     * a tree representing the formula is built from it.
+     * @param parsed Parsed input in MATA format.
+     * @return A vector of InterAutomata from each section in parsed input.
+     */
+    static std::vector<InterAutomaton> parse_from_mf(const Mata::Parser::Parsed& parsed);
+
+    bool states_enumerated() const {return state_naming == Naming::ENUM;}
```

Comment:
I don't get the naming of the function. Seems ambiguous. Maybe `are_states_enumerated_type`? Or something like that, since it could mean "states were enumerated", "states are enumerated", etc.

Reply:
would be `states_enum_type` ok?

Reply:
Better, or `states_are_enum_type`

Reply:
If the function returns `bool`, I would move the verb to the front: `are_states_enum_type`(?).

Reply:
But `if(nfa.states_are_enum_type())` sounds better :P

Reply:
I have no objections to it. I just like the convention of having the verb first. Nevertheless, use whichever feels more natural to you.

Reply:
rename to `are_x_enum_type()`